crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -442,10 +442,8 @@ impl Entities {
         // negative, meaning they are all new IDs to allocate, or a mix of both.
         let range_end = self
             .free_cursor
-            // Unwrap: these conversions can only fail on platforms that don't support 64-bit atomics
-            // and use AtomicIsize instead (see note on `IdCursor`).
-            .fetch_sub(IdCursor::try_from(count).unwrap(), Ordering::Relaxed);
-        let range_start = range_end - IdCursor::try_from(count).unwrap();
+            .fetch_sub(IdCursor::from(count), Ordering::Relaxed);
```

Comment:
Maybe this should use an expect with an error mentioning the lack of 64-bit atomics instead of just removing this information completely.